doc/contributing/coding_style_cpp.rst

```diff
@@ -41,3 +41,68 @@ Includes
 * Other includes should use absolute paths::
 
     #include <libdnf/.../*.hpp>
+
+Line breaks
+-----------
+clang-format doesn't enforce number of line breaks between top level definitions, hence the style of these is on a best-effort basis. Please try to adhere to the following conventions:
+
+* Single line break between top-level copyright comment, include guards and includes
+* Double line break before and between namespace blocks.
+* Single line break inside a namespace block at the start and end
+* Double line break between unrelated or loosely related definitions, single or no line break between related definitions
+* Single line break between a namespace block and an include guard #endif
+* Function/method definitions should be separated by a double line break
+
+Example::
+
+    // SPDX-License-Identifier: GPL-2.0-or-later
+    // SPDX-FileCopyrightText: Copyright contributors to the libdnf project
+
+    #ifndef LIBDNF_RPM_PACKAGE_SACK_HPP
+    #define LIBDNF_RPM_PACKAGE_SACK_HPP
+
+    #include "package.hpp"
+    #include "package_set.hpp"
+
+    #include "libdnf/base/base_weak.hpp"
+
+    #include <map>
+    #include <string>
+
+
+    namespace libdnf {
+
+    class Goal;
+
+    }  // namespace libdnf
```

Comment:
Shouldn't we make forward declarations more compact by omiting the empty lines?

Reply:
Feels like one line break at least is a good idea since we also don't indent into namespaces but I don't mind either.